src/context/likedContext.js

```diff
@@ -0,0 +1,87 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import { SET_ERROR, SET_LOADING, SET_SUCCESS } from "reducers/constants";
+import { sharedReducer } from "reducers/reducerFunctions";
+import { sharedState } from "reducers/states";
+import { useAuth } from "./authContext";
+
+const LikedContext = createContext();
+
+const LikedProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(sharedReducer, sharedState);
+  const { data: likedVideos, loading, error } = state;
+  const {
+    userAuthState: { authToken },
+  } = useAuth();
+
+  const getLikedVideos = async () => {
+    dispatch({ type: SET_LOADING });
+    try {
+      const response = await axios.get("/api/user/likes", {
+        headers: { authorization: authToken },
+      });
+      setTimeout(() => {
+        dispatch({
+          type: SET_SUCCESS,
+          payload: response.data.likes,
+        });
+      }, 3000);
```

Comment:
Why setTimeout() is used here?

Reply:
To make my api slower so that I can display my custom card loader animation, peacefully :rofl:

Reply:
Nvm, got it.. to show the loading on the likes page. Cool loading animation!